Rules/UseConsistentIndentation.cs

```diff
@@ -146,10 +148,30 @@ public override IEnumerable<DiagnosticRecord> AnalyzeScript(Ast ast, string file
                 switch (token.Kind)
                 {
                     case TokenKind.AtCurly:
-                    case TokenKind.AtParen:
-                    case TokenKind.LParen:
                     case TokenKind.LCurly:
+                        AddViolation(token, indentationLevel++, diagnosticRecords, ref onNewLine);
+                        break;
+
                     case TokenKind.DollarParen:
+                    case TokenKind.AtParen:
+                        lParenSkippedIndentation.Push(false);
+                        AddViolation(token, indentationLevel++, diagnosticRecords, ref onNewLine);
+                        break;
+
+                    case TokenKind.LParen:
+                        // When a line start with a parenthesis and it is not the last non-comment token of that line,
+                        // then indentation does not need to be increased.
+                        if ((tokenIndex == 0 || tokens[tokenIndex - 1].Kind == TokenKind.NewLine) &&
+                            (tokenIndex < tokens.Length - 1 && NextTokenIgnoringComments(tokens, tokenIndex)?.Kind != TokenKind.NewLine))
```

Comment:
The first check here is also done by the method, which encodes the check succeeding as returning a null, which is also dealt with. So the first check isn't required

Reply:
The method has this check where `startIndex` is the passed in `tokenIndex`. Isn't that different to the `tokenIndex < tokens.Length - 1` check?
```csharp
            if (startIndex == tokens.Length - 1)
            {
                return null;
            }
```
But I agree that logic like that should probably be moved into the `NextTokenIgnoringComments` method and simplified if possible.

Reply:
My read of it is that the external check is checking that `tokenIndex` is in the valid range, and that the internal check does the same thing (albeit slightly less safely; `startIndex == tokens.Length - 1` should be `startIndex >= tokens.Length - 1` ideally) and returns `null` otherwise

Reply:
simplified by moving external check into internal check (`startIndex >= tokens.Length - 1`) in b3bf36d